### ISSUE TITLE
out-of-memory when running the gcn sampling example

### ISSUE THREAD

**Issue by zhuoyw** [dmlc/dgl#846]
## To Reproduce

1. cd examples/pytorch/sampling
2. python **gcn_cv_sc.py** --dataset reddit-self-loop --dropout 0.2 **--num-neighbors 2** --lr 0.01 --weight-decay 0 --n-hidden 128 **--num-layers 2** --n-epochs 20
or
2. python **gcn_ns_sc.py** --dataset reddit-self-loop --dropout 0.2 **--num-neighbors 10** --lr 0.01 --weight-decay 0 **--n-hidden 128 ** --num-layers 1 --n-epochs 20
## Expected behavior

The above code should run without any problems.

## Environment

 - DGL Version (e.g., 1.0): master branch 0fb13f7
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): pytorch 1.2.0
 - OS (e.g., Linux): centos 7
 - How you installed DGL (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): python setup.py install
 - Python version: 3.6
 - CUDA/cuDNN version (if applicable): 9.0
 - GPU models and configuration (e.g. V100): k40
 - Any other relevant information:

## Additional context

For cv gcn, setting the number of gcn layer set to 1 is working, but 2 is not OK.
I know that with pre-processing step,  it means a 3-layer gcn. 
But it is still strange to see the out-of-CPU-memory error when I am using GPU.

The error occurs after the training of the first epoch. I suspect that the testset evaluation part is to blame.

Could you please help me with it?
